x-pack/plugins/maps/public/connected_components/map/mb/sort_layers.test.ts

```diff
@@ -0,0 +1,223 @@
+
+  constructor(style: MbStyle) {
+    this._style = _.cloneDeep(style);
+  }
+
+  getStyle() {
+    return _.cloneDeep(this._style);
+  }
+
+  moveLayer(id: string, beforeId?: string) {
+    moveCounter++;
+
+    if (!this._style.layers) {
+      throw new Error(`Can not move layer, mapbox style does not contain layers`);
+    }
+
+    const layerIndex = this._style.layers.findIndex((layer) => {
+      return layer.id === id;
+    });
+    if (layerIndex === -1) {
+      throw new Error(`Can not move layer, layer with id: ${id} does not exist`);
+    }
+    const moveMbLayer = this._style.layers[layerIndex];
+
+    if (beforeId) {
+      const beforeLayerIndex = this._style.layers.findIndex((mbLayer) => {
+        return mbLayer.id === beforeId;
+      });
+      if (beforeLayerIndex === -1) {
+        throw new Error(`Can not move layer, before layer with id: ${id} does not exist`);
+      }
+      this._style.layers.splice(beforeLayerIndex, 0, moveMbLayer);
+    } else {
+      const topIndex = this._style.layers.length;
+      this._style.layers.splice(topIndex, 0, moveMbLayer);
+    }
+
+    // Remove layer from previous location
+    this._style.layers.splice(layerIndex, 1);
+
+    return this;
+  }
+}
+
+class MockMapLayer {
+  private readonly _id: string;
+  private readonly _labelsOnTop: boolean;
+
+  constructor(id: string, labelsOnTop: boolean) {
+    this._id = id;
+    this._labelsOnTop = labelsOnTop;
+  }
+
+  ownsMbLayerId(mbLayerId: string) {
+    return mbLayerId.startsWith(this._id);
+  }
+
+  labelsOnTop() {
+    return this._labelsOnTop;
+  }
+
+  getId() {
+    return this._id;
+  }
+}
+
+test('getIsTextLayer', () => {
+  const paintLabelMbLayer = {
+    id: `mylayer_text`,
+    type: 'symbol',
+    paint: { 'text-color': 'red' },
+  } as MbLayer;
+  expect(getIsTextLayer(paintLabelMbLayer)).toBe(true);
+
+  const layoutLabelMbLayer = {
+    id: `mylayer_text`,
+    type: 'symbol',
+    layout: { 'text-size': 'red' },
+  } as MbLayer;
+  expect(getIsTextLayer(layoutLabelMbLayer)).toBe(true);
+
+  const iconMbLayer = {
+    id: `mylayer_text`,
+    type: 'symbol',
+    paint: { 'icon-color': 'house' },
+  } as MbLayer;
+  expect(getIsTextLayer(iconMbLayer)).toBe(false);
+
+  const circleMbLayer = { id: `mylayer_text`, type: 'circle' } as MbLayer;
+  expect(getIsTextLayer(circleMbLayer)).toBe(false);
+});
+
+describe('sortLayer', () => {
+  const ALPHA_LAYER_ID = 'alpha';
+  const BRAVO_LAYER_ID = 'bravo';
+  const CHARLIE_LAYER_ID = 'charlie';
+
+  const spatialFilterLayer = (new MockMapLayer(
+    SPATIAL_FILTERS_LAYER_ID,
+    false
+  ) as unknown) as ILayer;
+  const mapLayers = [
+    (new MockMapLayer(CHARLIE_LAYER_ID, true) as unknown) as ILayer,
+    (new MockMapLayer(BRAVO_LAYER_ID, false) as unknown) as ILayer,
+    (new MockMapLayer(ALPHA_LAYER_ID, false) as unknown) as ILayer,
+  ];
+
+  beforeEach(() => {
+    moveCounter = 0;
+  });
+
+  test('Should sort initial layer load order to expected order', () => {
```

Comment:
add comment that this is intermediate state between `ILayer.syncLayerWithMb` and `syncLayerOrder`.